2000-11-02  Jakub Jelinek  <jakub@redhat.com>

	* gcc.dg/20001101-1.c: New test.
	* gcc.dg/20001102-1.c: New test.

--- gcc/testsuite/gcc.dg/20001101-1.c.jj	Wed Nov  1 16:40:31 2000
+++ gcc/testsuite/gcc.dg/20001101-1.c	Wed Nov  1 16:40:27 2000
@@ -0,0 +1,38 @@
+/* { dg-do run { target sparc*-*-* } } */
+/* { dg-options "-O2 -mvis" } */
+
+int foo(double a, int b, int c, double *d, int h)
+{
+  int f, g;
+  double e;
+
+l:
+  f = (int) a;
+  a -= (double) f;
+  if (b == 1)
+    {
+      g = c;
+      f += g;
+      c -= g;
+    }
+  if (b == 2)
+    {
+      f++;
+      h = c;
+    }
+  if (!h)
+    {
+      for (g = 0; g <= 10; g++)
+        for (h = 0; h <= 10; h++)
+          e += d [10 + g - h];
+      goto l;
+    }
+  return f & 7;
+}
+
+int main()
+{
+  if (foo(0.1, 1, 3, 0, 1) != 3)
+    abort ();
+  exit (0);
+}
--- gcc/testsuite/gcc.dg/20001102-1.c.jj	Thu Aug 24 05:00:32 2000
+++ gcc/testsuite/gcc.dg/20001101-2.c	Thu Nov  2 06:16:37 2000
@@ -0,0 +1,39 @@
+/* { dg-do run { target sparc*-*-* } } */
+/* { dg-options "-O2 -mvis" } */
+
+int foo(double a, int b, int c, double *d, int h)
+{
+  int f, g;
+  double e;
+
+l:
+  f = (int) a;
+  a -= (double) f;
+  if (b == 1)
+    {
+      g = c;
+      f += g;
+      c -= g;
+    }
+  if (b == 2)
+    {
+      f++;
+      h = c;
+      goto l;
+    }
+
+  asm volatile ("" : : :
+		"f0", "f1", "f2", "f3", "f4", "f5", "f6", "f7",
+		"f8", "f9", "f10", "f11", "f12", "f13", "f14", "f15",
+		"f16", "f17", "f18", "f19", "f20", "f21", "f22", "f23",
+		"f24", "f25", "f26", "f27", "f28", "f29", "f30", "f31");
+
+  return f & 7;
+}
+
+int main()
+{
+  if (foo(0.1, 1, 3, 0, 1) != 3)
+    abort ();
+  exit (0);
+}
